src/callbacks/onLayout.js

```diff
@@ -1,65 +1,17 @@
-import { select } from 'd3';
-import backButton from './onLayout/backButton';
+import detailsLayout from './onLayout/detailsLayout';
+import controlGroupLayout from './onLayout/controlGroupLayout';
 import augmentFilters from './onLayout/augmentFilters';
 import augmentOtherControls from './onLayout/augmentOtherControls';
+import topXaxis from './onLayout/topXaxis';
 
 export default function onLayout() {
     const context = this;
 
-    //Move control labels and descriptions inside a div to display them vertically, label on top of description.
-    this.controls.wrap.selectAll('.control-group').each(function(d) {
-        const controlGroup = select(this),
-            label = controlGroup.select('.control-label'),
-            description = controlGroup.select('.span-description'),
-            container = controlGroup.append('div').classed('label-description', true);
-
-        controlGroup.attr('class', `${controlGroup.attr('class')} ${d.type}`);
-
-        container.node().appendChild(label.node());
-        container.node().appendChild(description.node());
-
-        //Add horizontal rule to group controls and filters.
-        if (d.value_col === context.config.id_col)
-            context.controls.wrap
-                .insert('div', ':first-child')
-                .classed('controls horizontal-rule', true)
-                .text('Controls');
-        else if (d.option === 'y.grouping') {
-            const filterRule = context.controls.wrap
-                .append('div')
-                .classed('filters horizontal-rule', true)
-                .text('Filters');
-            context.controls.wrap.node().insertBefore(filterRule.node(), this.nextSibling);
-        }
-    });
+    //Lay out details container.
```

Comment:
Really nice job modularizing all of this.

Reply:
Thanks!  Makes it a lot easier to follow the sequence of events.